r/R/dataset-scan.R

```diff
@@ -118,9 +113,6 @@ Scanner$create <- function(dataset,
   if (use_threads) {
     scanner_builder$UseThreads()
   }
-  if (use_async) {
```

Comment:
Is there an R equivalent of a deprecation warning we should be raising?

Reply:
@jonkeane thoughts?

Reply:
Yeah, we should warn about this. https://github.com/apache/arrow/blob/e64480db51fc9622d02613f3ec60bac34d765092/r/R/deprecated.R#L21 is an example of this.

Something like:
```
.Deprecated(msg = "Use of 'use_async' is deprecated")
```

With a bit more explanation if we're now totally ignoring it, or what

Reply:
Ok, I've added the fields back in with deprecation warnings.